internal/provider/network_data_source.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	netbird "github.com/netbirdio/netbird/management/client/rest"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &NetworkDataSource{}
+
+func NewNetworkDataSource() datasource.DataSource {
+	return &NetworkDataSource{}
+}
+
+// NetworkDataSource defines the data source implementation.
+type NetworkDataSource struct {
+	client *netbird.Client
+}
+
+func (d *NetworkDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_network"
+}
+
+func (d *NetworkDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Read Network Settings and Metadata, see [NetBird Docs](https://docs.netbird.io/how-to/networks) for more information.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Network ID",
+				Required:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Network Name",
+				Computed:            true,
+			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "Network Description",
+				Computed:            true,
+			},
+			"routers": schema.ListAttribute{
+				MarkdownDescription: "Network Router IDs",
+				ElementType:         types.StringType,
+				Computed:            true,
+			},
+			"resources": schema.ListAttribute{
+				MarkdownDescription: "Network Resource IDs",
+				ElementType:         types.StringType,
+				Computed:            true,
+			},
+			"policies": schema.ListAttribute{
+				MarkdownDescription: "Policy IDs associated with resources inside this Network",
+				ElementType:         types.StringType,
+				Computed:            true,
+			},
+			"routing_peers_count": schema.Int32Attribute{
```

Comment:
See [comment about *_count fields](https://github.com/netbirdio/terraform-provider-netbird/pull/1/files#r2081321210)

Reply:
Resolving in https://github.com/netbirdio/terraform-provider-netbird/pull/7